src/hotspot/share/opto/escape.hpp

```diff
@@ -429,21 +430,26 @@ class ConnectionGraph: public ResourceObj {
   int find_init_values_phantom(JavaObjectNode* ptn);
 
   // Set the escape state of an object and its fields.
-  void set_escape_state(PointsToNode* ptn, PointsToNode::EscapeState esc) {
+  void set_escape_state(PointsToNode* ptn, PointsToNode::EscapeState esc
+                        DEBUG_ONLY(COMMA const char* reason="reason unknown")) {
```

Comment:
Wouldn't it be better to not have a default argument for `reason` to enforce a unique message (even if it's `"reason unknown (42)"`) being set by the caller?

Same for `set_fields_escape_state` and `set_not_scalar_replaceable`.

Reply:
Yeah, that seems better. (didn't want to push that onto others initially). I'll drop the default reason string